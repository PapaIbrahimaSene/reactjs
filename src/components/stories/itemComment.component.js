import React from 'react';
import moment from 'moment';
import renderHTML from 'react-render-html';
import { Feed, Icon } from 'semantic-ui-react';

export const ItemComment = ({
  comment: { by, score, text, time, comments }
}) => (
  <Feed>
    <Feed.Event>
      <Feed.Label>
        <Icon name='user circle' color='orange'></Icon>
      </Feed.Label>
      <Feed.Content style={{fontSize: '90%'}}>
        <Feed.Summary>
          by <a>{ by }</a>
          <Feed.Date>
            { moment(time * 1000).calendar() }
          </Feed.Date>
        </Feed.Summary>
        <Feed.Extra text>
          { renderHTML(String(text)) }
        </Feed.Extra>
        <Feed.Extra text>
          { comments && comments.length > 0  && comments.map(comment => (
            <ItemComment key={ comment.id } comment={ comment }></ItemComment>
          ))}
        </Feed.Extra>
      </Feed.Content>
    </Feed.Event>
  </Feed>
);