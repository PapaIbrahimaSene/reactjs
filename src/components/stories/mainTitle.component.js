import React from 'react';
import { Container, Header, Icon } from 'semantic-ui-react'

export const MainTitle = () => (
    <Container text style={{ marginTop: '5em'}} >
    <Header as='h1' >
    <Icon inverted name = "hacker news" size = "big" color='red'/>
    News Stories and Comments From HackerNews
    <Header as='h6'>Top 10 News Stories and related - up to - 20 comments from Y Combinator's API</Header>
    </Header>
    </Container>
);

export default MainTitle;