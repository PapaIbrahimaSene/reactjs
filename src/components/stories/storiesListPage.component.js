import React from 'react';
import { MainMenu } from '../core/mainMenu.component';
import StoriesList from './storiesList.component';
import MainTitle from './mainTitle.component';

export const StoriesListPage = () => (
  <div>
    <MainMenu />
    <MainTitle />
    <StoriesList />
  </div>
);

export default StoriesListPage;