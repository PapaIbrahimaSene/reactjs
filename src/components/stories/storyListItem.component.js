import React from 'react';
import moment from 'moment';
import { connect } from 'react-redux';
import { Button, Feed, Icon, List } from 'semantic-ui-react';
import { ItemComment } from './itemComment.component';

const mapStateToProps = state => ({
  comment: state.comment
});

export class StoryListItem extends React.Component {
  state = {
    viewComment: false
  };

  handleToggleCommentsClick = () => {
    this.setState(prevState => Object.assign(prevState, {
      viewComment: !prevState.viewComment
    }));
  }

  render() {
    return (
      <List.Item>
        <List.Content floated='left'>
          <Feed>
            <Feed.Event>
              <Feed.Content>
                <Feed.Summary>
                  by <a>{ this.props.story.by }</a>
                  <Feed.Date>
                    { moment(this.props.story.time * 1000).calendar() }
                  </Feed.Date>
                </Feed.Summary>
                <Feed.Extra text >
                  <h4>
                    <a href={ this.props.story.url }
                      target='_blank'>
                      { this.props.story.title }
                    </a>
                  </h4>
                </Feed.Extra>
                <Feed.Meta>
                  <Feed.Like>
                  <Icon name='thumbs up outline' color='red' />
                  { this.props.story.score } Points &nbsp;&nbsp;&nbsp;
                  <Icon name='user circle' color='red' /> &nbsp;
                  { this.props.story.descendants || 'No'} Comments
                  </Feed.Like>
                </Feed.Meta>
              </Feed.Content>
            </Feed.Event>
          </Feed>
        </List.Content>
        <List.Content floated='right'>
            <Button
              basic color='red'
              style={{ fontSize: '80%', padding: '12px' }}
              disabled={ !this.props.story.descendants }
              onClick={ this.handleToggleCommentsClick }>
              { !this.state.viewComment && (<span>View Comments</span>) }
              { this.state.viewComment && (<span>Hide Comments</span>) }
              </Button>
        </List.Content>
        { this.state.viewComment && this.props.story.comments && this.props.story.comments.length > 0 &&
          <List.Content style={{ paddingTop: '12px' }} floated='left'>
            { this.props.story.comments.map(comment => (
              <ItemComment key={ comment.id } comment={ comment }></ItemComment>
            ))}
          </List.Content>
        }
      </List.Item>
    );
  }
}

export default connect(
  mapStateToProps
)(StoryListItem);
