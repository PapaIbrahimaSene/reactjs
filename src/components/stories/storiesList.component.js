import React from 'react';
import { connect } from 'react-redux';
import { StoryListItem } from './storyListItem.component';

import { fetchTopStories } from '../../selectors/stories.selector';

import {
  Container,
  Dimmer,
  List,
  Loader,
} from 'semantic-ui-react';

const mapDispatchToProps = (dispatch) => ({
  onLoadTopStories: () => dispatch(fetchTopStories())
});

const mapStateToProps = state => ({
  stories: state.stories,
});

export class StoriesList extends React.Component {
  state = {
    loading: true
  };

  componentDidMount() {
    this.props.onLoadTopStories()
      .then(() => this.setState(prevState => Object.assign(prevState, {
        loading: false
      })));
  }

  render() {
    return (
      <Container text style={{ marginTop: '5em' }}>
        { this.state.loading &&
          <Dimmer page active inverted>
            <Loader
              content='Loading'
              size='big' />
          </Dimmer>
        }
        <List relaxed='very' divided style={{ marginBottom: '24px' }}>
          {
            this.props.stories && this.props.stories.map(story => (
              <StoryListItem key={ story.id } story={ story } />
            ))
          }
        </List>
      </Container>
    );
  }
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(StoriesList);
