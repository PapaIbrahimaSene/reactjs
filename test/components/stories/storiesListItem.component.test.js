import React from 'react';
import { shallow } from 'enzyme';
import { StoryListItem } from '../../../src/components/stories/storyListItem.component';
import stories from '../../fixtures/stories';

test('should create StoryListItem', () => {
    const wrapper = shallow(<StoryListItem story={stories[0]} />);

    expect(wrapper).toMatchSnapshot();
});