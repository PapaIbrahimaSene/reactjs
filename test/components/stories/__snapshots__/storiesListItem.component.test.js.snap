// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create StoryListItem 1`] = `
<ListItem>
  <ListContent
    floated="left"
  >
    <Feed>
      <FeedEvent>
        <FeedContent>
          <FeedSummary>
            by 
            <a>
              alanfranzoni
            </a>
            <FeedDate>
              Today at 11:05 AM
            </FeedDate>
          </FeedSummary>
          <FeedExtra
            text={true}
          >
            <h4>
              <a
                href="https://imgur.com/a/RlUj9Ed"
                target="_blank"
              >
                NIST's answer to “Do you need a blockchain?”
              </a>
            </h4>
          </FeedExtra>
          <FeedMeta>
            <FeedLike
              as="a"
            >
              <Icon
                as="i"
                color="blue"
                name="thumbs up outline"
              />
              59
               Points    
              <Icon
                as="i"
                color="orange"
                name="user circle"
              />
                
              14
               Comments
            </FeedLike>
          </FeedMeta>
        </FeedContent>
      </FeedEvent>
    </Feed>
  </ListContent>
  <ListContent
    floated="right"
  >
    <Button
      as="button"
      basic={true}
      color="orange"
      disabled={false}
      onClick={[Function]}
      role="button"
      style={
        Object {
          "fontSize": "80%",
          "padding": "12px",
        }
      }
    >
      <span>
        View Comments
      </span>
    </Button>
  </ListContent>
</ListItem>
`;
